src/hotspot/share/ci/ciMethod.cpp

```diff
@@ -468,7 +468,7 @@ const BitMap& ciMethod::bci_block_start() {
 ciCallProfile ciMethod::call_profile_at_bci(int bci) {
   ResourceMark rm;
   ciCallProfile result;
-  if (method_data() != NULL && method_data()->is_mature()) {
+  if (method_data() != NULL && (method_data()->is_mature() || PolymorphicInlining)) {
```

Comment:
Is it ok with immature profiling data?

Reply:
Oh I think I made a mistake. Let's remove the condition and see if it impacts performance.